BTCPayServer/wwwroot/js/store-reports.js

```diff
@@ -0,0 +1,199 @@
+var app;
+var origData;
+
+srv.sortBy = function (field) {
+    for (let key in this.fieldViews) {
+        if (this.fieldViews.hasOwnProperty(key)) {
+            var sortedField = field == key;
+            var fieldView = this.fieldViews[key];
+
+            if (sortedField && (fieldView.sortBy === "" || fieldView.sortBy === "desc")) {
+                fieldView.sortByTitle = "asc";
+                fieldView.sortBy = "asc";
+                fieldView.sortIconClass = "fa fa-sort-alpha-asc";
+            }
+            else if (sortedField && (fieldView.sortByTitle === "asc")) {
+                fieldView.sortByTitle = "desc";
+                fieldView.sortBy = "desc";
+                fieldView.sortIconClass = "fa fa-sort-alpha-desc";
+            }
+            else {
+                fieldView.sortByTitle = "";
+                fieldView.sortBy = "";
+                fieldView.sortIconClass = "fa fa-sort";
+            }
+        }
+    }
+    this.applySort();
+}
+srv.applySort = function () {
+    var fieldIndex;
+    var fieldView;
+    for (let key in this.fieldViews) {
+        if (this.fieldViews.hasOwnProperty(key)) {
+            fieldView = this.fieldViews[key];
+            if (fieldView.sortBy !== "") {
+                fieldIndex = srv.result.fields.findIndex((a) => a.name === key);
+                break;
+            }
+            fieldView = null;
+        }
+    }
+    if (!fieldView)
+        return;
+    var sortType = fieldView.sortBy === "desc" ? 1 : -1;
+    srv.result.data.sort(function (a, b) {
+        var aVal = a[fieldIndex];
+        var bVal = b[fieldIndex];
+        if (aVal === bVal) return 0;
+        if (aVal === null) return 1 * sortType;
+        if (bVal === null) return -1 * sortType;
+        if (aVal > bVal) return 1 * sortType;
+        return -1 * sortType;
+    });
+};
+srv.dataUpdated = function () {
+    this.updateFieldViews();
+    origData = clone(this.result.data);
+    this.applySort();
+};
+srv.updateFieldViews = function () {
+    this.fieldViews = this.fieldViews || {};
+
+    // First we remove the fieldViews that doesn't apply anymore
+    for (let key in this.fieldViews) {
+        if (this.fieldViews.hasOwnProperty(key)) {
+            if (!this.result.fields.find(i => i.name === key))
+                delete this.fieldViews[key];
+        }
+    }
+
+    // Then we add those that are missing
+    for (let i = 0; i < this.result.fields.length; i++) {
+        const field = this.result.fields[i];
+        if (!this.fieldViews.hasOwnProperty(field.name)) {
+            this.fieldViews[field.name] =
+            {
+                sortBy: "",
+                sortByTitle: "",
+                sortIconClass: "fa fa-sort"
+            };
+        }
+    }
+};
+
+$(function () {
```

Comment:
reduce jquery usage:
```suggestion
document.addEventListener("DOMContentLoaded", () => {
```